components/core/src/clp/string_utils/string_utils.cpp

```diff
@@ -1,72 +1,72 @@
 #include "string_utils/string_utils.hpp"
 
 #include <algorithm>
-#include <charconv>
+#include <cctype>
 #include <cstring>
 
 using std::string;
 using std::string_view;
 
+namespace clp::string_utils {
 namespace {
 /**
- * Helper for ``wildcard_match_unsafe_case_sensitive`` to advance the pointer in
- * tame to the next character which matches wild. This method should be inlined
- * for performance.
- * @param tame_current
- * @param tame_bookmark
- * @param tame_end
- * @param wild_current
- * @param wild_bookmark
- * @return true on success, false if wild cannot match tame
+ * Helper for `wildcard_match_unsafe_case_sensitive` to advance `tame`'s iterator to the next
+ * character that matches the current character in `wild`, and to bookmark this character. If the
+ * current character in `wild` is escaped, `wild`'s iterator will also be advanced.
+ *
+ * NOTE:
+ * - This method expects that `tame_it` < `tame_end_it`
+ * - This method should be inlined for performance.
```

Comment:
I'm not sure if this is enforced. afaik we are just suggesting the compiler to inline the method. But I'm also not sure whether we should use `always_intline` attribute since we may need to compile this file using none-gnu compilers.

Reply:
It isn't enforced, but at the same time I don't know if *forcing* it to be inlined is necessary. The reason I said it should be inlined is because in past performance tests, the `inline` hint did make a difference. Nowadays though, it seems like gcc inlines it regardless of the hint.